galaxyproject/galaxy
# Display DOIs in Archived Histories

When a history has been archived preserving the data in an external RDM repository like Zenodo, we can display its [Document Object Identifier](https://www.doi.org/) on it.

![image](https://github.com/galaxyproject/galaxy/assets/46503462/55bc266c-6b80-41fe-bb9d-a1a9d8df2ceb)

<del>This PR also allows FilesSource plugins to serialize extra properties defined in the `serialize_extra_props` class variable. This adds flexibility to customize the information we want to serialize for the file source. In this particular case, we serialize to the `extra` fields the repository URL associated with the RDM file source so it can be used to fetch the DOI information.</del> We decided to replace the `extra` field with directly serializing the `url` for simplicity even if not all file sources might use it.

## How to test the changes?
(Select all options that apply)
  - Setup an RDM file source like in #18022
  - Archive a history creating an export record in that file source.
  - **[Important]** make sure **the record is published and has a DOI** (a fake one using the Zenodo Sandbox is fine but links won't resolve). 
  - The DOI should appear in your Archived Histories.

## License
